From 16e571e397b1045fc9a89b3851124c14a9255f87 Mon Sep 17 00:00:00 2001
From: Sumit Jamadar <xkbr38@motorola.com>
Date: Wed, 16 Sep 2015 16:24:02 -0500
Subject: [PATCH 158/959] IKMMINTG-6094: arm/dt: 8952: Add OV5693 camera
 bringup

Add ov5693 front facing camera support
for Athene 8952 platform.

Change-Id: Ia605c60385d4a6338d94a0f899be9d515f3da755
Signed-off-by: Sumit Jamadar <xkbr38@motorola.com>
Reviewed-on: http://gerrit.mot.com/785143
SME-Granted: SME Approvals Granted
SLTApproved: Slta Waiver <sltawvr@motorola.com>
Tested-by: Jira Key <jirakey@motorola.com>
Reviewed-by: Satya Prasad Kottisa <fvgm76@motorola.com>
Reviewed-by: Elena Satraitis <w04888@motorola.com>
Reviewed-by: Christopher Fries <cfries@motorola.com>
Submit-Approved: Jira Key <jirakey@motorola.com>
---
 .../boot/dts/qcom/msm8952-camera-sensor-moto.dtsi  | 90 ++++++++++++++++++++++
 arch/arm/boot/dts/qcom/msm8952-moto-common.dtsi    | 13 ++++
 2 files changed, 103 insertions(+)
 create mode 100644 arch/arm/boot/dts/qcom/msm8952-camera-sensor-moto.dtsi

diff --git a/arch/arm/boot/dts/qcom/msm8952-camera-sensor-moto.dtsi b/arch/arm/boot/dts/qcom/msm8952-camera-sensor-moto.dtsi
new file mode 100644
index 00000000000..74894d89fdb
--- /dev/null
+++ b/arch/arm/boot/dts/qcom/msm8952-camera-sensor-moto.dtsi
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2015, The Linux Foundation. All rights reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 and
+ * only version 2 as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+&cci {
+	qcom,camera@1 {
+		cell-index = <1>;
+		compatible = "qcom,camera";
+		qcom,csiphy-sd-index = <1>;
+		qcom,csid-sd-index = <1>;
+		qcom,mount-angle = <270>;
+		cam_vdig-supply = <&pm8950_l23>;
+		cam_vio-supply = <&pm8950_l10>;
+		qcom,cam-vreg-name = "cam_vdig", "cam_vio";
+		qcom,cam-vreg-min-voltage = <1200000 1800000>;
+		qcom,cam-vreg-max-voltage = <1200000 1800000>;
+		qcom,cam-vreg-op-mode = <1200000 1800000>;
+		pinctrl-names = "cam_default", "cam_suspend";
+
+		pinctrl-0 = <&cam_sensor_mclk2_default
+			&cam_sensor_front1_default>;
+		pinctrl-1 = <&cam_sensor_mclk2_sleep
+			&cam_sensor_front1_sleep>;
+
+		gpios = <&msm_gpio 28 0>,
+			<&msm_gpio 40 0>;
+		qcom,gpio-reset = <1>;
+		qcom,gpio-req-tbl-num = <0 1>;
+		qcom,gpio-req-tbl-flags = <1 0>;
+		qcom,gpio-req-tbl-label = "CAMIF_MCLK2",
+					"CAM_RESET2";
+		qcom,sensor-position = <1>;
+		qcom,sensor-mode = <0>;
+		qcom,cci-master = <0>;
+		status = "ok";
+		clocks = <&clock_gcc clk_mclk2_clk_src>,
+				<&clock_gcc clk_gcc_camss_mclk2_clk>;
+		clock-names = "cam_src_clk", "cam_clk";
+	};
+};
+
+&soc {
+	tlmm_pinmux: pinctrl@1000000 {
+
+		cam_sensor_front1_default {
+			/* RESET, STANDBY */
+			qcom,pins = <&gp 40>;
+			qcom,num-grp-pins = <1>;
+			label = "cam_sensor_front1_default";
+			/* active state */
+			cam_sensor_front1_default: cam_sensor_front1_default {
+				drive-strength = <2>; /* 2 MA */
+				bias-disable; /* No PULL */
+			};
+		};
+
+		cam_sensor_front1_sleep {
+			/* RESET, STANDBY */
+			qcom,pins = <&gp 40>;
+			qcom,num-grp-pins = <1>;
+			label = "cam_sensor_front1_sleep";
+			/*suspended state */
+			cam_sensor_front1_sleep: cam_sensor_front1_sleep {
+				drive-strength = <2>; /* 2 MA */
+				bias-disable; /* No PULL */
+			};
+		};
+	};
+};
+
+&rpm_bus {
+	rpm-regulator-ldoa10 {
+		status = "okay";
+		pm8950_l10: regulator-l10 {
+			regulator-min-microvolt = <1800000>;
+			regulator-max-microvolt = <2800000>;
+			qcom,init-voltage = <1800000>;
+			status = "okay";
+		};
+	};
+};
diff --git a/arch/arm/boot/dts/qcom/msm8952-moto-common.dtsi b/arch/arm/boot/dts/qcom/msm8952-moto-common.dtsi
index c798e5d9c93..fbed1cd4111 100644
--- a/arch/arm/boot/dts/qcom/msm8952-moto-common.dtsi
+++ b/arch/arm/boot/dts/qcom/msm8952-moto-common.dtsi
@@ -13,6 +13,7 @@
 
 #include "msm8952.dtsi"
 #include "msm8952-pinctrl.dtsi"
+#include "msm8952-camera-sensor-moto.dtsi"
 
 / {
 	aliases {
@@ -411,6 +412,18 @@
 			qcom,msm-mi2s-tx-lines = <1>;
 		};
 	};
+
+	cci: qcom,cci@1b0c000 {
+		pinctrl-names = "cci_default", "cci_suspend";
+			pinctrl-0 = <&cci0_active>;
+			pinctrl-1 = <&cci0_suspend>;
+		gpios = <&msm_gpio 29 0>,
+			<&msm_gpio 30 0>;
+		qcom,gpio-tbl-num = <0 1 >;
+		qcom,gpio-tbl-flags = <1 1>;
+		qcom,gpio-tbl-label = "CCI_I2C_DATA0",
+				"CCI_I2C_CLK0";
+	};
 };
 
 &sdhc_1 {
-- 
2.11.0

